source/ada/lsp-ada_contexts.adb

```diff
@@ -488,28 +488,47 @@ package body LSP.Ada_Contexts is
 
    procedure Get_Any_Symbol
      (Self        : Context;
-      Prefix      : VSS.Strings.Virtual_String;
+      Pattern     : LSP.Search.Search_Pattern_Access;
       Only_Public : Boolean;
       Callback : not null access procedure
         (File : GNATCOLL.VFS.Virtual_File;
          Name : Libadalang.Analysis.Defining_Name;
          Stop : in out Boolean))
    is
-      procedure Adapter
+      function Get_Defining_Name
         (File : GNATCOLL.VFS.Virtual_File;
-         Loc  : Langkit_Support.Slocs.Source_Location;
-         Stop : in out Boolean);
-      --  Find a Defining_Name at the given location Loc in a unit of URI and
-      --  pass it to Callback procedure call.
+         Loc  : Langkit_Support.Slocs.Source_Location)
+         return Libadalang.Analysis.Defining_Name;
+      --  Find a Defining_Name at the given location Loc in a unit of File
+
+      procedure Adapter
+        (File          : GNATCOLL.VFS.Virtual_File;
+         Defining_Name : Libadalang.Analysis.Defining_Name;
+         Stop          : in out Boolean);
+      --  Pass Defining_Name to Callback procedure call.
 
       -------------
       -- Adapter --
       -------------
 
       procedure Adapter
+        (File          : GNATCOLL.VFS.Virtual_File;
```

Comment:
this Adapter has no sense any more. You can pass `Callback` instead of `Adapter'Access`, right?

Reply:
done